### DISCUSSION TITLE
Image Loading Not Working

### DISCUSSION THREAD

**Discussion by zMidair** [ocornut/imgui#7986] [New Users Build/Link/Run issues ONLY!]
### Version/Branch of Dear ImGui:

Version 1.91.0

### Back-ends:

imgui-cocos

### Compiler, OS:

Windows 11 + MSVC 2022

### Details:

Hello!
I am trying to load an image from memory but I am failing and I don't know why. It isn't showing anything on the window.

### Minimal, Complete and Verifiable Example code:

`ImageUtils.hpp` (header I made for loading images easier)
```cpp
#pragma once
#include <imgui-cocos.hpp>

namespace ImageUtils
{
    bool LoadTextureFromMemory(unsigned char* image_data, size_t data_size, GLuint* out_texture, int* out_width, int* out_height)
    {
        // Load from file
        int image_width = 0;
        int image_height = 0;
        
        GLuint image_texture;
        glGenTextures(1, &image_texture);
        glBindTexture(GL_TEXTURE_2D, image_texture);

        // Setup filtering parameters for display
        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);

        // Upload pixels into texture
        glPixelStorei(GL_UNPACK_ROW_LENGTH, 0);
        glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, image_width, image_height, 0, GL_RGBA, GL_UNSIGNED_BYTE, image_data);

        *out_texture = image_texture;
        *out_width = image_width;
        *out_height = image_height;

        return true;
    }
}
```

`GUI.cpp` (draw func)
```cpp
// UI Elements
GLuint bannerTexture = 0;
int banner_width = 0, banner_height = 0;
ImageUtils::LoadTextureFromMemory(bannerRawData, sizeof(bannerRawData), &bannerTexture, &banner_width, &banner_height);

ImGui::Image((void*)(uintptr_t)bannerTexture, ImVec2(banner_width, banner_height));
```

`banner.c` (image)
```cpp
unsigned char bannerRawData[149807] = { 0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A, 0x00, 0x00, 0x00, 0x0D, ................ };
```

**Comment by ocornut** [comment]
I moved your issue to a discussion. Your code is incorrect and your question isn't really an imgui question. People here may be able to help.

Compare your code to the one on this page for OpenGL:
https://github.com/ocornut/imgui/wiki/Image-Loading-and-Displaying-Examples

- It's unclear what the format of bannerRawData[] is.
- Your function appears to be aiming to load raw data but I don't think 149807 would be the size of a raw image (it's more likely to be the size of a compressed image). Yet the code doesn't include any image format parsing, and is never writing image_width, image_height so those values are always zero.

**Reply by zMidair** [reply]
Hello,
I got a part of my code from there but the issue is that I don’t have that stb library, do I need to install it? (I’m modding cocos2dx)
